models/agent_helpers.go

```diff
@@ -655,11 +656,34 @@ func updateExternalExporterParams(q *reform.Querier, row *Agent) error {
 	return nil
 }
 
-func isPushMetricsSupported(pmmAgentVersion *string) bool {
+// IsPushMetricsSupported return if PUSH mode is supported for pmm agent version.
+func IsPushMetricsSupported(pmmAgentVersion *string) bool {
 	if agentVersion, err := version.Parse(pointer.GetString(pmmAgentVersion)); err == nil {
 		if agentVersion.Less(PMMAgentWithPushMetricsSupport) {
 			return false
 		}
 	}
 	return true
 }
+
+// SupportedMetricsMode automatically pick metrics mode.
+func SupportedMetricsMode(q *reform.Querier, metricsMode managementpb.MetricsMode, pmmAgentID string) (managementpb.MetricsMode, error) {
+	if metricsMode != managementpb.MetricsMode_AUTO {
```

Comment:
let's add before this
```
if pmmAgentID == PMMServerAgentID && metricsMode == managementpb.MetricsMode_PUSH {
    return errors.Errorf("push metrics mode is not allowed for exporters running on pmm-server")
}
```

Reply:
Case added. Also added this method in rds.